test/blackbox/common/RTPSBlackboxTestsBasic.cpp

```diff
@@ -1182,6 +1182,99 @@ TEST(RTPS, participant_ignore_local_endpoints_two_participants)
     eprosima::fastrtps::rtps::RTPSDomain::removeRTPSParticipant(participant_reader);
 }
 
+/* Maximum number of bytes allowed for an RTPS datagram generated by this participant. */
+TEST(RTPS, max_output_message_size_participant)
+{
+    /* Set up */
+    // Create the RTPSParticipants with the appropriate value for the property
+    auto testTransport =  std::make_shared<eprosima::fastdds::rtps::test_UDPv4TransportDescriptor>();
+    const uint32_t segment_size = 1470;
+    std::string segment_size_str = std::to_string(segment_size);
+    testTransport->messages_filter_ = [segment_size](eprosima::fastrtps::rtps::CDRMessage_t& datagram)
+            {
+                EXPECT_LE(datagram.length, segment_size);
+                return false;
+            };
+
+    // Create the RTPSReader
+    RTPSWithRegistrationReader<Data1mbPubSubType> reader(TEST_TOPIC_NAME);
+    reader.init();
+    EXPECT_TRUE(reader.isInitialized());
+
+    eprosima::fastrtps::rtps::RTPSParticipantAttributes patt;
+    patt.useBuiltinTransports = false;
+    patt.userTransports.push_back(testTransport);
+    patt.properties.properties().emplace_back("fastdds.max_message_size", segment_size_str);
+    eprosima::fastrtps::rtps::RTPSParticipant* participant_writer =
+            eprosima::fastrtps::rtps::RTPSDomain::createParticipant(static_cast<uint32_t>(GET_PID()) % 230, patt);
+    ASSERT_NE(participant_writer, nullptr);
+
+    // Create the RTPSWriter
+    RTPSWithRegistrationWriter<Data1mbPubSubType> writer(TEST_TOPIC_NAME, participant_writer);
+    writer.init();
+    EXPECT_TRUE(writer.isInitialized());
+
+    // Wait for discovery
+    writer.wait_discovery(1, std::chrono::seconds(2));
+    reader.wait_discovery(1, std::chrono::seconds(2));
+    EXPECT_EQ(writer.get_matched(), 1u);
+    EXPECT_EQ(reader.get_matched(), 1u);
+
+    // Send samples
+    auto samples = default_data16kb_data_generator(1);
+    reader.expected_data(samples);
+    reader.startReception();
+    writer.send(samples);
+    EXPECT_TRUE(samples.empty());
+
+    // Wait for reception
+    reader.block_for_all(std::chrono::seconds(1));
+    EXPECT_EQ(reader.getReceivedCount(), 1u);
+
+    /* Tear-down */
+    eprosima::fastrtps::rtps::RTPSDomain::removeRTPSParticipant(participant_writer);
+}
+
+/* Maximum number of bytes allowed for an RTPS datagram generated by this writer. */
+TEST(RTPS, max_output_message_size_writer)
+{
+    const uint32_t segment_size = 1470;
+    std::string segment_size_str = std::to_string(segment_size);
+
+    auto testTransport = std::make_shared<eprosima::fastdds::rtps::test_UDPv4TransportDescriptor>();
+    testTransport->messages_filter_ = [segment_size](eprosima::fastrtps::rtps::CDRMessage_t& datagram)
+            {
+                EXPECT_LE(datagram.length, segment_size);
+                return false;
```

Comment:
Let's remind future readers what `return false;` means here:

```suggestion
                // Never drop samples
                return false;
```